src/contributors/MetricsTableContributor.ts

```diff
@@ -152,8 +159,81 @@ export class MetricsTableContributor extends Contributor {
     /** @override */
     /** todo !!!! specify data type and return type  */
     public computeData(data: Array<MetricsTableResponse>): MetricsTable {
-        console.log('test is ok');
-        return null;
+        // todo: to be improved
+        const headers = new Map();
+        const rows: Map<string, MetricsTableRow> = new Map();
+        const maxCount = new Map();
+        let rowDataMaxLength = 0;
+
+        data = this.orderMetricsTableResponse(data);
+        data.forEach(metricsResponse => {
+            metricsResponse.aggregationResponse.elements.forEach(elements => {
+                elements.metrics.forEach(metrics => {
+                    const uniqColumn = `${metricsResponse.collection}_${metrics.field}_${metrics.type}`;
+                    const uniqKeyForSum = `${elements.key_as_string}_${metrics.type}`;
+                    // storing headers;
+                    if(!headers.has(uniqColumn)) {
+                        const headerItem: MetricsTableHeader = {
+                            title: metricsResponse.collection ,
+                            subTitle: metrics.field,
+                            metric: metrics.type
+                        };
+                        headers.set(uniqColumn, headerItem);
+                    }
+
+                    // storing uniq row;
+                    if(rows.has(uniqKeyForSum)) {
+                        rows.get(uniqKeyForSum).data.push({maxValue: 0, value: metrics.value});
+                    } else {
+                        const metricsTableData: MetricsTableData = {maxValue: 0, value: metrics.value};
+                        const metricsTableRow: MetricsTableRow = {data: [], term: elements.key_as_string};
+                        metricsTableRow.data.push(metricsTableData);
+                        rows.set(uniqKeyForSum, metricsTableRow);
```

Comment:
the unicity of the row should be based on the key_as_string only, I don't see why we should concatenate the type